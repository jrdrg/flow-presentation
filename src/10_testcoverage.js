// @-flow

import React from 'react';

/*
type ButtonsProps = {
    value: { amount: number },
    incrementValue: (value: number) => void
};
*/

export const Buttons = (props) => {
    return (
        <div>
            <button
                onClick={() => props.incrementValue(props.value.amount + 1)}
            >
                click me
            </button>
        </div>
    );
};


export const ButtonsWrapper = (props) => {
    return (
        <div>
            <Buttons value={1} />
        </div>
    );
};
