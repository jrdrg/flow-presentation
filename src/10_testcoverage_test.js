import React from 'react';
import {Buttons, ButtonsWrapper} from './10_testcoverage';
import ShallowRenderer from 'react-test-renderer/shallow';

describe('100% test coverage!!! woohoo!', () => {

    it('should render Buttons in the ButtonWrapper', () => {
        const shallowRenderer = new ShallowRenderer();
        shallowRenderer.render(<ButtonsWrapper />);
        const wrapper = shallowRenderer.getRenderOutput();

        expect(wrapper).toMatchSnapshot();
    });


    it('should call onClick', () => {
        let incremented = null;

        const onClick = value => {
            incremented = value;
        };

        const value = { amount: 1 };
        const shallowRenderer = new ShallowRenderer();
        shallowRenderer.render(
            <Buttons
                value={value}
                incrementValue={onClick}
            />
        );
        const buttons = shallowRenderer.getRenderOutput();
        const buttonOnClick = buttons.props.children.props.onClick;

        buttonOnClick();

        expect(incremented).toBe(2);
    });
});
