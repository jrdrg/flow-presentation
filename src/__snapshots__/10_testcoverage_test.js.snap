// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`100% test coverage!!! woohoo! should render Buttons in the ButtonWrapper 1`] = `
<div>
  <Unknown
    value={1}
  />
</div>
`;
